### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by pcooksey** [svrvsm/dca#2]
Given inputs:
`9.5,15.3,20.5,21.7,39.7,40.3,41.3,350,355,360`

Your program outputs:
`16.8,355.0`

Do you agree with output? I believe your assumption is less than 10 degrees between points are combined into one cluster.

**Comment by vsankarl** [comment]
I agree with the output.

There are two conditions that need to be met for the data points to form a cluster. 
1. Within 10 degrees of separation
2. Mean - outermost_feature > 3. If it is less than 3 then it's likely the object is located perpendicular to the view mirror and requires further inspection to identify its true center. Alternatively, the object can also be tall and narrow. Ideally, this needs to be tracked separately and evaluated using "r" values. For this problem, it is skipped. The basis for this constraint is identified using input2.txt - data points 0.2,1.2,3.1,3.2,4.1,5.9 are ignored though they are within 10 degrees of separation. 

For the above input 
 
9.5,15.3,20.5,21.7, ==> Forms a cluster with center 16.8
39.7,40.3,41.3, ==> Constraint 2 failed, likely not a cluster. Require more examination ( outside the scope of the problem ) 
350,355,360 ==> Forms a cluster with center 355.0

Here is the log.txt ( gets generated in the current directory) excerpt

======= Processing ./cus.txt ===========
Info: Distance 5.8 Cur 15.3 prev 9.5
Info: Distance 5.2 Cur 20.5 prev 15.3
Info: Distance 1.2 Cur 21.7 prev 20.5
Info: Distance 18 Cur 39.7 prev 21.7
 >>>>>>>> Found cluster center at 16.75 using sum 67, range: feature[ 0 ] = 9.5 ==> feature[ 3 ] = 21.7 deltaFromMean 7.25 
Info: Distance 0.599998 Cur 40.3 prev 39.7
Info: Distance 1 Cur 41.3 prev 40.3
Info: Distance 308.7 Cur 350 prev 41.3
 Info: Skipping potentially small width (or less visible) cluster 39.7,40.3,41.3
 Info: Mean 40.4333 Min feature index 4 value 39.7 delta 0.733334
Info: Distance 5 Cur 355 prev 350
Info: Distance 5 Cur 360 prev 355
 >>>>>>>> Found cluster center at 355 using sum 1065, range: feature[ 7 ] = 350 ==> feature[ 9 ] = 360 deltaFromMean 5 
 ======= Success - centers 16.8,355.0 =======

**Comment by pcooksey** [comment]
I agree with your assumptions (and I had seen and appreciated the nice logging!) Let me be more specific aren't  `9.5` and `360` actually within 10 degrees of separation?

**Comment by vsankarl** [comment]
Ah, you got me :)  Code needs to handle that case.   

Now input2.txt,  { 0.2,1.2,3.1,3.2,4.1,5.9}  and { 350.3,353.5,359.5,359.8 }  makes sense. The entire set form a single cluster.
